lxd/instances_post.go

```diff
@@ -705,6 +717,15 @@ func createFromBackup(d *Daemon, r *http.Request, projectName string, data io.Re
 			return fmt.Errorf("Load instance: %w", err)
 		}
 
+		// Generate a new UUID for the instance when importing it from backup.
+		// We will later use this new instance UUID as a VM Generation ID
+		// in order to differentiate the two instances (before and after backup)
```

Comment:
Can you mention "changes that instance's place in time (moving it backwards)" somewhere in here please.

Reply:
Similarly to my comment above, I took a close look at this and I think we can do a bit better.

Rather than running another DB query to update the instance record after its been created as part of the import, we should modify the config that is applied to the DB as the instance record is created.

To do this I think we can modify `internalImportFromBackup` as its commend say:

```go
// internalImportFromBackup creates instance, storage pool and volume DB records from an instance's backup file.
```

And the `volatile.uuid` key is in the instance's DB record (restored from the instance's backup file), so this seems like the perfect place to intercept it.

The call to `instance.CreateInternal` inside `internalImportFromBackup` is where the instance's DB record is created.
So modifying the `instDBArgs.Config` between `backup.ConfigToInstanceDBArgs` and `instance.CreateInternal` should do it.